### PR TITLE
fix(app): configure modules during calibration, shorten heater shaker fixture name

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

At the beginning of the module calibration flow, the user is asked to locate the module on the deck.
This integrated the deck configurator component directly into this location selction step of the
module calibration wizard. the selected location will now be saved directly to deck configuration.

Closes [RQA-2603](https://opentrons.atlassian.net/browse/RQA-2603)

# Review requests

- Run module calibration and confirm that the selected location reflects the deck configuration

# Risk assessment
low

[RQA-2603]: https://opentrons.atlassian.net/browse/RQA-2603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/organisms/ModuleWizardFlows/PlaceAdapter.tsx [hunk 1]
```diff
@@ -62,7 +63,10 @@ export const PlaceAdapter = (props: PlaceAdapterProps): JSX.Element | null => {
   const { t } = useTranslation('module_wizard_flows')
   const mount = attachedPipette.mount
   const cutoutId = deckConfig.find(
-    cc => cc.opentronsModuleSerialNumber === attachedModule.serialNumber
+    cc =>
+      cc.opentronsModuleSerialNumber === attachedModule.serialNumber &&
+      (attachedModule.moduleType !== THERMOCYCLER_MODULE_TYPE ||
+        cc.cutoutFixtureId === THERMOCYCLER_V2_FRONT_FIXTURE)
```
Comment (CaseyBatten) [inline app/src/organisms/ModuleWizardFlows/PlaceAdapter.tsx:92]: This fix looks appropriate and solves the issues we were having in the case of multiple thermocycler fixtures proccing on the front end from the deck config.

**Review by CaseyBatten — APPROVED** [review]
Overall looks solid, app flow makes sense when using and I was able to successfully calibrate a thermocycler on a Flex with no interruptions.

File: components/src/hardware-sim/BaseDeck/BaseDeck.tsx [hunk 2]
```diff
@@ -101,7 +103,12 @@ export function BaseDeck(props: BaseDeckProps): JSX.Element {
   const singleSlotFixtures = deckConfig.filter(
     fixture =>
       fixture.cutoutFixtureId != null &&
-      SINGLE_SLOT_FIXTURES.includes(fixture.cutoutFixtureId)
+      (SINGLE_SLOT_FIXTURES.includes(fixture.cutoutFixtureId) ||
+        // If module fixture is loaded, still visualize singleSlotFixture underneath for consistency
+        Object.entries(MODULE_FIXTURES_BY_MODEL).reduce<CutoutFixtureId[]>(
+          (acc, [_model, fixtures]) => [...acc, ...fixtures],
+          []
+        ))
```
Comment (brenthagen) [inline components/src/hardware-sim/BaseDeck/BaseDeck.tsx]: this renders single slot fixtures in (under) every cutout...maybe this is ok?

Screen Shot 2024-04-19 at 3 47 58 PM

Comment (b-cooper) [reply components/src/hardware-sim/BaseDeck/BaseDeck.tsx]: interesting..., yeah maybe it should only be done for module fixtures. Good catch! I'll patch it up

**Review by brenthagen — APPROVED** [review]
LGTM !

**Pull request merged**